wally-registry-backend/src/auth.rs

```diff
@@ -47,6 +66,34 @@ fn match_api_key<T>(request: &Request<'_>, key: &str, result: T) -> Outcome<T, a
     }
 }
 
+async fn verify_github_token(request: &Request<'_>) -> Outcome<WriteAccess, anyhow::Error> {
+    let token: String = match request.headers().get_one("authorization") {
+        Some(key) if key.starts_with("Bearer ") => (key[6..].trim()).to_owned(),
+        _ => {
+            return Outcome::Failure((Status::Unauthorized, format_err!("Github auth required")));
+        }
+    };
+
+    let client = reqwest::Client::new();
+    let response = client
+        .get("https://api.github.com/user")
+        .header("accept", "application/json")
+        .header("user-agent", "wally")
+        .bearer_auth(token)
+        .send()
+        .await
+        .unwrap()
```

Comment:
Let's map this error into the appropriate failure status `Outcome` instead of unwrapping. We don't wanna crash the registry if the GitHub API goes down

Reply:
Oh yeah definitely!